cdap-cli/src/main/java/co/cask/cdap/cli/command/view/CreateOrUpdateViewCommand.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.cli.command.view;
+
+import co.cask.cdap.api.data.format.FormatSpecification;
+import co.cask.cdap.api.data.format.Formats;
+import co.cask.cdap.api.data.schema.Schema;
+import co.cask.cdap.cli.ArgumentName;
+import co.cask.cdap.cli.CLIConfig;
+import co.cask.cdap.cli.ElementType;
+import co.cask.cdap.cli.english.Article;
+import co.cask.cdap.cli.english.Fragment;
+import co.cask.cdap.cli.util.AbstractAuthCommand;
+import co.cask.cdap.cli.util.ArgumentParser;
+import co.cask.cdap.client.ViewClient;
+import co.cask.cdap.internal.io.SchemaTypeAdapter;
+import co.cask.cdap.proto.Id;
+import co.cask.cdap.proto.ViewConfig;
+import co.cask.common.cli.Arguments;
+import com.google.common.base.Joiner;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonObject;
+import com.google.inject.Inject;
+
+import java.io.IOException;
+import java.io.PrintStream;
+import java.util.Collections;
+import java.util.Map;
+
+/**
+ * Creates or updates a view.
+ */
+public class CreateOrUpdateViewCommand extends AbstractAuthCommand {
+
+  private static final Gson GSON = new GsonBuilder()
+    .registerTypeAdapter(Schema.class, new SchemaTypeAdapter())
+    .create();
+  private final ViewClient client;
+
+  @Inject
+  public CreateOrUpdateViewCommand(ViewClient client, CLIConfig cliConfig) {
+    super(cliConfig);
+    this.client = client;
+  }
+
+  @Override
+  public void perform(Arguments arguments, PrintStream output) throws Exception {
+    Id.View viewId = Id.View.from(cliConfig.getCurrentNamespace(), arguments.get(ArgumentName.VIEW.toString()));
+    Id.Stream streamId = Id.Stream.from(cliConfig.getCurrentNamespace(), arguments.get(ArgumentName.STREAM.toString()));
+    String formatName = arguments.get(ArgumentName.FORMAT.toString());
+    Schema schema = getSchema(arguments);
+
+    Map<String, String> settings = Collections.emptyMap();
+    if (arguments.hasArgument(ArgumentName.SETTINGS.toString())) {
+      settings = ArgumentParser.parseMap(arguments.get(ArgumentName.SETTINGS.toString()));
+    }
+    FormatSpecification formatSpecification = new FormatSpecification(formatName, schema, settings);
```

Comment:
Appears to be so. Most of this class uses the same logic as https://github.com/caskdata/cdap/blob/develop/cdap-cli/src/main/java/co/cask/cdap/cli/command/SetStreamFormatCommand.java (didn't bother to move the code to a common place since we're going to remove it anyway)